students/bbirsoz1/Module_3/mailroom1.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+import sys  # imports go at the top of the file
+
+#Mailroom Part 1
+
+donor_db = [("William Gates, III", [653772.32, 12.17]),
+            ("Jeff Bezos", [877.33]),
+            ("Paul Allen", [663.23, 43.87, 1.32]),
+            ("Mark Zuckerberg", [1663.23, 4300.87, 10432.0]),
+            ("Obama", [99999, 3333]),
+            ]
+
+main_menu_prompt = "\n".join(("Welcome to the Mailroom Charity Menu!",
+                    "Please choose from below options:",
+                    "1 - Send a Thank You",
+                    "2 - Create a Report",
+                    "3 - Quit",
+                    ">>> "))
+
+
+thank_menu_prompt = "\n".join(("Please enter one of the following:",
+                    "1 - Enter a new thank you note",
+                    "2 - List of donors in the database",
+                    "3 - Return to main menu",
+                    '>>> '))
+
+def list_of_donors():
+          print("List of donors in the database:")
+          for donor in donor_db:
+              print(donor)          
+
+def create_a_report():
+    print("Report:")
+    Header = ('Donor Name', 'Total Donation', "Number of Donations", 'Average Donation')
+    print("-"*75)
+    print("{:<20} | {:^24} | {:^19} | {:^15}".format(Header[0], Header[1], Header[2], Header[3]))
+    print("-"*75)
+    '''
+    everything = [ ]
+    donor_db.sort(key=sort_donation, reverse=True)
+    for i, j in donor_db():
+        total_donation = (sum(j))
+        num_of_donation = (len(j))
+        average_donation = (sum(j) / len(j))
+        everything.append([i, total_donation, num_of_donation, average_donation])
+    print("-"*75)
+    I can't figure this out
+    '''
```

Comment:
Hi Begum,

For _create_a_report()_ try using more descriptive variables while iterating over the list of donors, that will make it easier for you to understand what your code was doing if you look at it later on. Also, you use use a similar print statement to the one you used for the headers to print each donor's data:

```
    for donor_name, donations in donor_db:
        total_donation = sum(donations)
        num_of_donation = len(donations)
        average_donation = int(total_donation / num_of_donation)
        print("{:<20} | {:>24} | {:^19} | {:>15}".format(donor_name, total_donation, num_of_donation, average_donation))
```
Regards,

Luis